src/admin.tsx

```diff
@@ -1,30 +1,101 @@
+      setData((snapshot.docs.map(doc => doc.data())));
+    });
 
-  return (
+    return unsubscribe;
+  }, [db]);
 
-    <div id="admin-page" data-testid='admin'>
-      <h1>Admin Page</h1>
-      <p>Welcome {currentUser?.email}</p>
-
-      <div className="login-button-parent">
-        <Button
-          sx={{ '&:hover': { bgcolor: 'white' } }}
-          variant="contained"
-          size="large"
-          onClick={signOut}
+
+  const displayTreeComponent = () => {
+    if (!data) {
+      return;
+    }
+    // Group by subcategory
+    const newData: Record<string, DocumentData> = {};
+    data?.forEach((category) => {
+      const groupedVideos: Record<string, DocumentData> = {};
+      category.videos.forEach((vid: DocumentData) => {
+        if (groupedVideos[vid.group]) {
+          groupedVideos[vid.group] = groupedVideos[vid.group].concat(vid);
+        } else {
+          groupedVideos[vid.group] = [vid];
+        }
+      });
+
+      newData[category.name] = groupedVideos;
+    })
+
+    const elements = Object.entries(newData).map(([key, groupedVideos]) => (
+      <TreeItem className='tree-item' key={key} nodeId={key} label={key} sx={{textAlign: 'left'}}>
+        {Object.entries(groupedVideos).map(([subcategory, vids]) => (
+          <TreeItem key={key + subcategory} nodeId={key + subcategory} label={subcategory} sx={{textAlign: 'left'}}>
+            {(vids as any).map((vid: DocumentData) => (
+              <TreeItem key={vid.name} nodeId={vid.name} label={vid.name} sx={{textAlign: 'left'}} />
+            ))}
+          </TreeItem>
+        ))}
+      </TreeItem>
+    ));
+
+    return (
+      <div id='tree-view'>
+        <TreeView
+          aria-label="file system navigator"
+          defaultCollapseIcon={<ExpandMoreIcon />}
+          defaultExpandIcon={<ChevronRightIcon />}
+          sx={{ height: '100%', flexGrow: 1, maxWidth: 600, overflowY: 'auto', paddingLeft: 10 }}
         >
-          Sign Out
-        </Button>
+          {elements}
+        </TreeView>
       </div>
+    )
+  }
 
-    </div>
 
+  return (
+    <div id="admin-page" data-testid='admin'>
+      <Box sx={{ flexGrow: 1 }}>
+        <AppBar className="app-bar" >
+          <Toolbar>
+            <Typography variant="h4" component="div" sx={{ flexGrow: 1, textAlign: 'left' }}>
+              <b><i>Welcome,</i></b> {currentUser?.email?.split('@')[0]}
+            </Typography>
+            <ButtonGroup variant='outlined' size='large' sx={{ bgcolor: 'black'}}>
+              <Button>Analytics</Button>
+              <Button onClick={() => document.getElementById('videos-title')?.scrollIntoView(true)}>Videos</Button>
+              <Button onClick={signOut}>Logout</Button>
+            </ButtonGroup>
+          </Toolbar>
+        </AppBar>
+      </Box>
+
+      <Typography id="videos-title" variant='h1'>videos</Typography>
+      {displayTreeComponent()}
+
+    </div>
   );
-} 
\ No newline at end of file
+}
+
+
+
```

Comment:
Nit: Can you remove all this whitespace? Should be able to setup VSCode to auto do this on save if you want to